internal/pkg/skuba/kubernetes/node_version.go

```diff
@@ -221,26 +221,38 @@ func allWorkerNodesTolerateVersionWithVersioningInfo(allNodesVersioningInfo Node
 	return true
 }
 
-// AllControlPlanesMatchVersion checks that all control planes are on the same version, and that they match the current cluster version
-func AllControlPlanesMatchVersion(currentClusterVersion *version.Version) (bool, error) {
+// AllControlPlanesMatchVersion checks that all control planes are on the same version, and that they match a cluster version
+func AllControlPlanesMatchVersion(clusterVersion *version.Version) (bool, error) {
 	allNodesVersioningInfo, err := AllNodesVersioningInfo()
 	if err != nil {
 		return false, err
 	}
-	return AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo, currentClusterVersion), nil
+	return AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo, clusterVersion), nil
 }
 
-// AllControlPlanesMatchVersionWithVersioningInfo checks that all control planes are on the same version, and that they match the current cluster version
+// AllControlPlanesMatchVersionWithVersioningInfo checks that all control planes are on the same version, and that they match a cluster version
 // With the addition of passing in a NodeVersionInfoMap to make it easily testable
-func AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo NodeVersionInfoMap, currentClusterVersion *version.Version) bool {
+func AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo NodeVersionInfoMap, clusterVersion *version.Version) bool {
 	for _, nodeInfo := range allNodesVersioningInfo {
 		// skip workers
 		if !nodeInfo.IsControlPlane() {
 			continue
 		}
-		if !nodeInfo.ToleratesClusterVersion(currentClusterVersion) {
+		if !nodeInfo.ToleratesClusterVersion(clusterVersion) {
 			return false
 		}
 	}
 	return true
 }
+
+// AllNodesMatchVersionWithVersioningInfo returns if all nodes match a specific kubernetes version
+// This can be used to determine e.g. if an upgrade is ongoing
+func AllNodesMatchClusterVersionWithVersioningInfo(allNodesVersioningInfo NodeVersionInfoMap, clusterVersion *version.Version) (bool, error) {
+	match := true
+	for _, version := range allNodesVersioningInfo {
+		if !version.EqualsClusterVersion(clusterVersion) {
+			match = false
+		}
+	}
+	return match, nil
+}
```

Comment:
This method can be simplified so we don't iterate always over all elements in case that any of them does not match. Also, no need to add error to the signature if we are not generating one inside or bubbling up.

```go
func AllNodesMatchClusterVersionWithVersioningInfo(allNodesVersioningInfo NodeVersionInfoMap, clusterVersion *version.Version) bool {
	for _, version := range allNodesVersioningInfo {
		if !version.EqualsClusterVersion(clusterVersion) {
			return false
		}
	}
	return true
}
```